worldedit-core/src/main/java/com/fastasyncworldedit/core/function/mask/CachedMask.java

```diff
@@ -11,8 +11,8 @@
 public class CachedMask extends AbstractDelegateMask implements ResettableMask {
 
     private final boolean hasExtent;
-    private transient LocalBlockVectorSet cache_checked = new LocalBlockVectorSet();
```

Comment:
Isn't the trick behind the CachedMask that the set is reset when needed? That code should be revisited then too, and usages of the class should be revisited too to make sure that the additional overhead here isn't worse than not caching at all.

Reply:
It will always be faster than re-testing a mask at any particular point and the memory overhead is very low if even only a few  coordinate points are used as it's ultimately backed by a SparseBitSet

Reply:
This would also be API-breaking as BlockVectorSet's index is based on pairing x and z coords, and does not include y coord.

Reply:
> It will always be faster than re-testing a mask at any particular point

Well that depends on the actual mask. I didn't dig through the usages, but that's why I mentioned it. If all usages are fine, I'm happy with it.

Reply:
Yeah it's only used either in operations that are likely to go back on themselves a lot, or masks that check the blocks around frequently. This leads to cross-chunk-boundary checks that can increase operation time as the same chunk could be loaded/unloaded from FAWE multiple times in an operation (based on the size of the edit, and the parallel-threads/target-size settings).